association-backend/src/association_membership/domain/entities.py

```diff
@@ -30,23 +42,17 @@ class Meta(BaseMeta):
         tablename = "subscriptions"
 
     id: int = ormar.Integer(primary_key=True)
-    customer: Customer = ormar.ForeignKey(Customer, nullable=False)
-    stripe_subscription_id: str = ormar.String(nullable=False, max_length=256)
+    user_id: int = ormar.Integer(unique=True)
     status: SubscriptionStatus = ormar.String(
         max_length=20,
         choices=list(SubscriptionStatus),
         default=SubscriptionStatus.PENDING,
         nullable=False,
     )
 
-    _add_invoice: List[SubscriptionInvoice] = PrivateAttr(default_factory=list)
-
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        super().__init__(*args, **kwargs)
-        self._add_invoice = []
-
-    def add_invoice(self, payment: SubscriptionInvoice):
-        self._add_invoice.append(payment)
+    _add_stripe_subscription_payment: List[StripeSubscriptionPayment] = PrivateAttr(
```

Comment:
What's the purpose of this private attribute? it's the Orm that wants it? Seems unnecessary otherwise...

Reply:
see here: https://github.com/pythonitalia/pycon/pull/2553/files/6fea3cf05683082eac8f6fe54300b3074a707da5#diff-700c143ae661c70592ba9db5004cf76292fc6b55efbf54ff109adcc1be15f846R97

this is used to store what objects should be added when saving the object in the repository.

This allows us to separate DB operations from Domain operations